backend/StreamServerApp/views.py

```diff
@@ -11,13 +11,21 @@ def index(request):
     return render(request, "index.html")
 
 def get_videos(request):
-    qs = Video.objects.all()
-    qs_json = serializers.serialize('json', qs)
-    return HttpResponse(qs_json, content_type='application/json')
+    if request.method == 'GET':
+        query = request.GET.get('folder', '')
+        if query == '':
+            qs = Video.objects.all()
+            qs_json = serializers.serialize('json', qs)
+            return HttpResponse(qs_json, content_type='application/json')
+        else:
+            qs = Folder.objects.filter(path=query)
+            print(qs)
+            videos = qs[0].videos.all()
```

Comment:
You can maybe use:
`qs.first().videos.all()`
instead of qs[0]

Also, I think you may want to do a _select related_ when doing the query on folders.
You should check, but it is possible that you are doing two queries here.
One when you get the video, and one when you serialize them.
Maybe adding the select related will get all the data in a single query.

I guess you get the first video for now, and you plan on doing something else.
I think the code is not robust, what happens if there is no video retrieved?